handsontable/src/plugins/base/base.js

```diff
@@ -211,13 +219,39 @@ export class BasePlugin {
    */
   onUpdateSettings(newSettings) {
     if (this.isEnabled) {
+      const hasEntryInSettings = ((settings) => {
```

Comment:
Can you move that logic to the private method within the `Base` class? I think that IIFE obscures the body of the method as well the function expression is not necessarily created and executed each time the settings are changed.